src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -310,6 +313,93 @@ public void terminatesMonitorsAndProcessIfClosedNormal() throws Exception {
     }
 
     /**
+     * Runs a java process which will throw a stack trace and makes sure it
+     * will group the stack trace.
+     */
+    @Test
+    public void logCompleteStackTrace() {
+        final org.apache.log4j.Logger logger =
+                org.apache.log4j.Logger.getRootLogger();
+        final TestAppender appender = new TestAppender();
+        logger.addAppender(appender);
+        final String[] commands = new String[] {
+            retrieveJavaExecLocation(), "-cp",
+            System.getProperty("java.class.path"),
+            "com.jcabi.log.VerboseProcessExample",
+        };
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        VerboseProcess process = null;
+        IllegalArgumentException caught = null;
+        try {
+            process = new VerboseProcess(builder, Level.INFO, Level.SEVERE);
+            process.stdout();
+        } catch (final IllegalArgumentException ex) {
+            caught = ex;
+        } finally {
+            logger.removeAppender(appender);
+            if (process != null) {
+                process.close();
+            }
+        }
+        Assert.assertNotNull(caught);
+        MatcherAssert.assertThat(
+                caught.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_1)
+        );
+        MatcherAssert.assertThat(
+                caught.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_2)
+        );
+        verifyLogs(appender);
+    }
+
+    /**
+     * Gets the location of Java, whether on Linux of Windows.
+     * @return String with Java location
+     */
+    public static String retrieveJavaExecLocation() {
```

Comment:
@dean-e-clark `public static` methods are not allowed, see https://github.com/teamed/qulice/issues/659